README.md

```diff
@@ -2,92 +2,707 @@
+- `put(lpath, rpath, recursive=False, **kwargs)` - Upload local file(s) as remote object(s)
+- `get(rpath, lpath, recursive=False, **kwargs)` - Download remote object(s) to local file(s)
+
+##### Utility Methods
+
+- `get_access_data()` - Get cache access statistics
+- `info(path, **kwargs)` - Get detailed information about an object
+- `exists(path, **kwargs)` - Check if a path exists
+- `isfile(path, **kwargs)` - Check if a path is an object
+- `isdir(path, **kwargs)` - Check if a path is a collection
+
+### OSDFFileSystem
+
+Convenience class that automatically connects to the OSDF federation (`osg-htc.org`).
+
+```python
+from pelicanfs.core import OSDFFileSystem
+
+# Equivalent to PelicanFileSystem("pelican://osg-htc.org")
+osdf = OSDFFileSystem()
+```
+
+### PelicanMap
+
+Create a filesystem mapper for use with libraries like xarray.
+
+```python
+PelicanMap(root, pelfs, check=False, create=False)
+```
+
+**Parameters:**
+- `root` (str): The root path in the Pelican namespace
+- `pelfs` (PelicanFileSystem): An initialized PelicanFileSystem instance
+- `check` (bool, optional): Check if the path exists. Default: `False`
+- `create` (bool, optional): Create the path if it doesn't exist. Default: `False`
+
+## Examples
+
+### Repository Examples
+
+See the `examples/` directory for complete working examples:
+
+- `examples/pelicanfs_example.ipynb` - Basic PelicanFS usage
+- `examples/pytorch/` - Using PelicanFS with PyTorch for machine learning
+- `examples/xarray/` - Using PelicanFS with xarray for scientific data
+- `examples/intake/` - Using PelicanFS with Intake catalogs
+
+### Project Pythia OSDF Cookbook
+
+For comprehensive tutorials and real-world geoscience examples, see the [Project Pythia OSDF Cookbook](https://projectpythia.org/osdf-cookbook/), which includes:
+
+- **NCAR GDEX datasets**: Meteorological, atmospheric composition, and oceanographic observations
+- **FIU Envistor**: Climate datasets from south Florida
+- **NOAA SONAR data**: Fisheries datasets in Zarr format
+- **AWS OpenData**: Sentinel-2 satellite imagery
+- **Interactive notebooks**: All examples are runnable in Binder or locally
+
+The cookbook demonstrates streaming large scientific datasets using PelicanFS with tools like xarray, Dask, and more.
+
+## Troubleshooting
+
+### Common Issues
+
+**Problem:** `NoAvailableSource` error when trying to access a file
+
+**Solution:** This usually means no cache or origin is available for the namespace. Check:
+- The namespace path is correct
+- The federation URL is correct
+- Network connectivity to the federation
+- Try enabling `direct_reads=True` to bypass caches
+
+**Problem:** `401 Unauthorized` or authorization errors
+
+**Solution:**
+- Ensure you've provided a valid token via the `headers` parameter
+- Check that your token has the correct scopes for the operation
+- Verify the token hasn't expired
+
+**Problem:** Slow performance
+
+**Solution:**
+- Try specifying `preferred_caches` to use a cache closer to you
```

Comment:
I don't think this is good advice -- there are too many factors beyond geolocation that go into whether a cache is performant, and it's beyond what a user can piece together on their own. This line would trigger Miron's wrath if he read it haha